### PR TITLE
Catalyst round info integration 

### PR DESCRIPTION
**Pull request by neuodev**
The algorithm used in this PR

```
if `currentFund` is available {
 if it is before `registrationStart` {
  show 'round {currentFund.id} registration starts at {registrationStart}'
 }
 if it is after `registrationEnd` {
  if it is before `votingStart` {
    show "Registration has ended. Voting starts at {votingStart}"
  }
  if it is between `votingStart` and `votingEnd` {
   show "Registration has ended.  Voting ends at  {votingEnd}"
  }
  if it is after `votingEnd` {
    show "Round {currentFund.id} ended.
    if `nextFund` is available {
      show "Round {nextFund.id} starts at {nextFund.registrationStart}"
    }
  }
 }
} else { // `currentFund` is not available
 if nextFund is available {
   show "Round {nextFund.id} starts at {nextFund.registrationStart}"
 } else {
   show a generic "Catalyst Round Info unavailable please check the Catalyst app for more info" page with no specific info
 }

```

### CODE REVIEW

File: packages/yoroi-extension/app/stores/ada/VotingStore.js [hunk 1]
```diff
@@ -110,12 +111,27 @@ export default class VotingStore extends Store<StoresMap, ActionsMap> {
...
   }
 
   get isActionProcessing(): boolean {
     return this.progressInfo.stepState === StepState.PROCESS;
   }
 
+  @action _getCatalystRoundInfo: void => Promise<void> = async () => {
+    const publicDeriver = this.stores.wallets.selected;
+    if (!publicDeriver) {
+      return
```
Comment (yushih) [inline packages/yoroi-extension/app/stores/ada/VotingStore.js]: Please end this and the below lines with a semicolon.

File: packages/yoroi-extension/app/containers/wallet/voting/VotingPage.js [hunk 2]
```diff
@@ -23,23 +23,56 @@ import type { TokenInfoMap } from '../../../stores/toplevel/TokenInfoStore';
...
 type Props = {|
   ...InjectedOrGenerated<GeneratedData>,
 |};
 
-const roundInfo = {
-  startDate: new Date(Date.parse('2021-07-08T16:00:00Z')),
-  endDate: new Date(Date.parse('2021-07-19T11:00:00Z')),
-  nextRound: 5,
-};
+export const messages: * = defineMessages({
```
Comment (yushih) [inline packages/yoroi-extension/app/containers/wallet/voting/VotingPage.js]: no need for `export`

File: packages/yoroi-extension/app/containers/wallet/voting/VotingPage.js [hunk 3]
```diff
@@ -102,24 +135,93 @@ export default class VotingPage extends Component<Props> {
...
-                ? roundInfo.nextRound
-                : roundInfo.nextRound - 1
-            }
+          <RegistrationOver
+            title={intl.formatMessage(messages.unavailableTitle)}
+            subtitle={intl.formatMessage(messages.unavailableSubtitle)}
           />
-        );
-      }
+          )
```
Comment (yushih) [inline packages/yoroi-extension/app/containers/wallet/voting/VotingPage.js]: Please fix the indentation and end the statement with a semicolon. Also apply to the below lines.

File: packages/yoroi-extension/app/containers/wallet/voting/VotingPage.js [hunk 4]
```diff
@@ -102,24 +135,93 @@ export default class VotingPage extends Component<Props> {
...
+                subtitle={intl.formatMessage(messages.earlyForRegistrationSubTitle, {
+                  roundNumber: currentFund.id,
+                  registrationStart: currentFund.registrationStart
+                })}
+              />
+              )
+            }
+
+            // registeration is ended -> check for voting start and end dates
+            if (true || isLate) {
```
Comment (yushih) [inline packages/yoroi-extension/app/containers/wallet/voting/VotingPage.js]: Please check the condition

**Review by yushih — CHANGES_REQUESTED** [review]
Please fix the issues.

And there is one possible improvement: currently, if the user navigates to the voting page before catalyst round info query is returned, because at this time `VotingStore.catalystRoundInfo` is `undefined`, the voting page would show "Catalyst Round Info unavailable", and then it would change when the query is returned. It is better to show something like "Fetching Catalyst info..." instead.

File: packages/yoroi-extension/app/containers/wallet/voting/VotingPage.js [hunk 5]
```diff
@@ -102,24 +135,93 @@ export default class VotingPage extends Component<Props> {
...
-      }
+          )
+        }
+        const { currentFund, nextFund } = catalystRoundInfo;
+        const nextFundRegistrationSubtitle = intl.formatMessage(messages.nextFundRegistration, {
+          roundNumber: nextFund?.id,
+          registrationStart: nextFund?.registrationStart
+        })
+
+        if(currentFund) {
```
Comment (yushih) [inline packages/yoroi-extension/app/containers/wallet/voting/VotingPage.js]: space between `if` and `(` to keep the style consistent

File: packages/yoroi-extension/app/containers/wallet/voting/VotingPage.js (hunk 3 above)
Comment (neuodev) [reply packages/yoroi-extension/app/containers/wallet/voting/VotingPage.js]: Okay

File: packages/yoroi-extension/app/containers/wallet/voting/VotingPage.js (hunk 4 above)
Comment (neuodev) [reply packages/yoroi-extension/app/containers/wallet/voting/VotingPage.js]: Sorry 😄😄
I was testing that this case will give the expected response

File: packages/yoroi-extension/app/containers/wallet/voting/VotingPage.js [hunk 6]
```diff
@@ -102,24 +135,100 @@ export default class VotingPage extends Component<Props> {
...
-
     // keep enabled on the testnet
+    const { catalystRoundInfo, loadingCatalystRoundInfo } = this.generated.stores.substores.ada.votingStore;
+    if (loadingCatalystRoundInfo) {
+      return (
+        <VerticallyCenteredLayout>
+         <LoadingSpinner />
+      </VerticallyCenteredLayout>
+      );
+    }
```
Comment (neuodev) [inline packages/yoroi-extension/app/containers/wallet/voting/VotingPage.js]: @yushih  I added the loading state 👌

File: packages/yoroi-extension/app/containers/wallet/voting/VotingPage.js [hunk 7]
```diff
@@ -102,24 +135,100 @@ export default class VotingPage extends Component<Props> {
...
-                ? roundInfo.nextRound
-                : roundInfo.nextRound - 1
-            }
+          <RegistrationOver
+            title={intl.formatMessage(messages.unavailableTitle)}
+            subtitle={intl.formatMessage(messages.unavailableSubtitle)}
           />
         );
       }
+        const { currentFund, nextFund } = catalystRoundInfo;
```
Comment (yushih) [inline packages/yoroi-extension/app/containers/wallet/voting/VotingPage.js]: Please fix indentation from here to line 231.

**Review by yushih — DISMISSED** [review]
Some indentations need to be fixed. Otherwise LGTM.

File: packages/yoroi-extension/app/containers/wallet/voting/VotingPage.js (hunk 7 above)
Comment (neuodev) [reply packages/yoroi-extension/app/containers/wallet/voting/VotingPage.js]: Done

**Review by yushih — APPROVED** [review]
@AhmedIbrahim336 there's still an indentation error that I've fixed. Please check indentations after you are done.

**Review by yushih — APPROVED** [review]

**Pull request merged**